destination/destination.go

```diff
@@ -99,16 +99,34 @@ func (d *Destination) Open(ctx context.Context) error {
 // Write routes incoming records to their appropriate handler based on the
 // operation.
 func (d *Destination) Write(ctx context.Context, recs []sdk.Record) (int, error) {
-	for i, r := range recs {
-		// TODO send all queries to postgres in one round-trip
-		err := sdk.Util.Destination.Route(ctx, r,
-			d.handleInsert,
-			d.handleUpdate,
-			d.handleDelete,
-			d.handleInsert,
-		)
+	b := &pgx.Batch{}
+	for _, rec := range recs {
+		var err error
```

Comment:
we should move this line outside the loop 👀

Reply:
Why though? 🤔 This error is only relevant inside the for loop.

Reply:
hmm my reasoning was to not create a new variable each time in the loop, and because when err is not nil we will return from the method, so no need to have a new empty err each time

Reply:
I expect the Go compiler to optimize this. You made me curious so I created two simple Go programs, the only difference between them being the place where the variable is declared. The compiler actually produced the same assembly code for both programs. Of course, I can't say _for sure_ this is the case here, but I sure expect that it is.

That said, _even if_ the produced assembly code is different, I'd be surprised if it made a noticeable difference. And given that we're writing code for humans and not for machines, I put more importance on writing clear and understandable code, which includes limiting the scope of variables to avoid accidental reuse.

Reply:
oh interesting! thank you for taking this extra step and checking the assembly code.
yeah I agree it's definitely more understandable too